environment.py

```diff
@@ -142,6 +142,11 @@ def env() -> Mapping[str, Optional[str]]:
         #
         'azul_current_catalog': None,
 
+        # A space-separated list of the sources in the current catalog to
+        # operate on, or "*" for all sources in the catalog.
```

Comment:
Would `azul_current_sources="*foo* *bar*" make reindex` work? If so, we may need to modify this comment a bit.

Reply:
No, it fails because none of the globs match:
```
$ azul_current_sources="*foo* *bar*" make reindex
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
python /home/nadove/PycharmProjects/azul-sc/scripts/check_branch.py
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
python scripts/reindex.py --delete --index --purge
envhook.py: Warning: azul_current_sources is '*foo* *bar*' but should be '*', you must run `source environment`
2024-05-31 18:09:23,354   DEBUG MainThread __main__: Source glob '*bar*' matched sources [] in catalog 'dcp3'
2024-05-31 18:09:23,355   DEBUG MainThread __main__: Source glob '*foo*' matched sources [] in catalog 'dcp3'
2024-05-31 18:09:23,355   DEBUG MainThread __main__: Source glob '*bar*' matched sources [] in catalog 'lungmap'
2024-05-31 18:09:23,355   DEBUG MainThread __main__: Source glob '*foo*' matched sources [] in catalog 'lungmap'
2024-05-31 18:09:23,355   DEBUG MainThread __main__: Source glob '*bar*' matched sources [] in catalog 'lm2'
2024-05-31 18:09:23,355   DEBUG MainThread __main__: Source glob '*foo*' matched sources [] in catalog 'lm2'
2024-05-31 18:09:23,355 WARNING MainThread __main__: Source(s) not found in any catalog: {'*bar*', '*foo*'}
Traceback (most recent call last):
  File "/home/nadove/PycharmProjects/azul-sc/scripts/reindex.py", line 206, in <module>
    main(sys.argv[1:])
  File "/home/nadove/PycharmProjects/azul-sc/scripts/reindex.py", line 150, in main
    require(any(sources_by_catalog.values()),
  File "/home/nadove/PycharmProjects/azul-sc/src/azul/__init__.py", line 1612, in require
    reject(not condition, *args, exception=exception)
  File "/home/nadove/PycharmProjects/azul-sc/src/azul/__init__.py", line 1631, in reject
    raise exception(*args)
azul.RequirementError: No valid sources specified for any catalog
make: *** [Makefile:128: reindex] Error 1
```

Reply:
It would work if at least one source contained either "foo" or "bar" as a substring.

Reply:
```suggestion
        # A space-separated list of shell globs to match the sources in the current catalog against, or "*" for all sources in the catalog. Only matching sources will be operated on. If no sources match, an error will be raised.
```